x-pack/plugins/security_solution/public/detections/components/alerts_table/timeline_actions/investigate_in_resolver.tsx

```diff
@@ -5,23 +5,36 @@
  * 2.0.
  */
 
+import { useMemo } from 'react';
 import { get } from 'lodash/fp';
 import type { EcsSecurityExtension as Ecs } from '@kbn/securitysolution-ecs';
+import { useIsExperimentalFeatureEnabled } from '../../../../common/hooks/use_experimental_features';
+
+export const useIsInvestigateInResolverActionEnabled = (ecsData?: Ecs) => {
+  const sentinelOneDataInAnalyzerEnabled = useIsExperimentalFeatureEnabled(
+    'sentinelOneDataInAnalyzerEnabled'
+  );
+  const [isEndpointOrSysmonFromWinlogBeat, hasProcessEntityId] = useMemo(() => {
+    const fileBeatModules = [
+      ...(sentinelOneDataInAnalyzerEnabled ? ['sentinel_one_cloud_funnel', 'sentinel_one'] : []),
+    ] as const;
+
+    const agentType = get(['agent', 'type', 0], ecsData);
+    const processEntityIds = get(['process', 'entity_id'], ecsData);
+    const firstProcessEntityId = get(['process', 'entity_id', 0], ecsData);
+    const eventModule = get(['event', 'module', 0], ecsData);
+    const eventDataStream = get(['event', 'dataset'], ecsData);
+    const datasetIncludesSysmon =
+      Array.isArray(eventDataStream) &&
+      eventDataStream.some((datastream) => datastream.includes('windows.sysmon'));
+    const agentTypeIsEndpoint = agentType === 'endpoint';
+    const agentTypeIsWinlogBeat = agentType === 'winlogbeat' && eventModule === 'sysmon';
+    const agentTypeIsFilebeat = agentType === 'filebeat' && fileBeatModules.includes(eventModule);
+    return [
```

Comment:
Just for readability, could we set those conditions in two constants `const constA = ...; const constB = ...` and then `return [constA, constB]`, as it was before? For me it seems hard to read an array of two items with a set of logical conditions. Let me know what do you think.

Reply:
Sure, will do 👍